### PR TITLE
PLF-6840 Session timeout not working reliably

### PR DESCRIPTION
**Pull request by trongtt**
Use a rest service to get the list of online users instead of a portlet request which keep session alive.

### CODE REVIEW

File: component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java [hunk 1]
```diff
@@ -53,24 +66,32 @@
...
     }
 
     @GET
-    @Path("isusersessionalive")
+    @Path("onlinefriends")
     @Produces(MediaType.APPLICATION_JSON)
-
-    public Response isUserSessionAlive() throws Exception {
+    @RolesAllowed("users")
+    public Response onlineFriends() throws Exception {
```
Comment (thomasdelhomenie) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: This class aims to expose services related to general platform information. It does not look very good to me to have this endpoint here. I think we should move it to https://github.com/exodev/commons/blob/develop/commons-component-common/src/main/java/org/exoplatform/services/user/RESTUserService.java. WDYT ?

Comment (trongtt) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: Yes, I was thinking about this and we should indeed.

File: component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java [hunk 2]
```diff
@@ -204,6 +225,42 @@ public String getPlatformEdition() {
...
+            Collections.reverse(users);
+            if (users.size() > MAX_USER) {
+                users = users.subList(0, INDEX_USER);
+            }
+
+            User userOnLine = null;
+
+            for (UserStateModel userModel : users) {
+                String user = userModel.getUserId();
+                String superUserName = System.getProperty("exo.super.user");
```
Comment (trongtt) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: (Y)

File: component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java (hunk 1 above)
Comment (trongtt) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: @thomasdelhomenie I find that this depends on services from Social (IdentityManager and LinkProvider), so we can not move it to _commons_ project.

I agree that this class seems to be served for general platform information, but it actually has only one endpoint **/info** here and being used in Platform applications only. Do you think we should just keep it here, OR we could rename this class/endpoints to better meaning names ?

Comment (thomasdelhomenie) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: Is moving RESTUserService class to platform project an option ? I don't know if the RESTUserService endpoints are used in others projects than platform.

Comment (trongtt) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: It's hard to say 100% sure BUT from what I see:
- RESTUserService introduces many rest endpoints BUT only `/rest/state/ping` is being declared in ECMS project code.
- It is declared in ECMS BUT only being invoked in Platform project.

Apparently, there is no problem in moving these parts (RESTUserService class and code in ECMS) to platform project. It makes sense to be there indeed.

The situation is quite messy for now, do you think we would take a risk (quite low I think) to make the change now ?

Comment (thomasdelhomenie) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: As you say, the risk seems low. I prefer to do this change than to have related endpoints spread over multiple locations and make things harder to understand and maintain.
WDYT ?

Comment (trongtt) [inline component/common/src/main/java/org/exoplatform/platform/common/rest/PlatformInformationRESTService.java]: Yes, let's do it best.

File: extension/webapp/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 3]
```diff
@@ -60,4 +60,15 @@
       <module>es6promise</module>
     </depends>
   </module>
+
+  <module>
+    <name>plf-utils</name>
```
Comment (thomasdelhomenie) [inline extension/webapp/src/main/webapp/WEB-INF/gatein-resources.xml]: This sounds very generic. We should name it more accurately IMO. What about "presence" ?

Comment (trongtt) [inline extension/webapp/src/main/webapp/WEB-INF/gatein-resources.xml]: My idea is that, the current `sendPing` method is just an utility which could be used in any place. We could add more utility methods there if any (probably there is).

WDYT ?

Comment (thomasdelhomenie) [inline extension/webapp/src/main/webapp/WEB-INF/gatein-resources.xml]: Yes I understand but I am not a big fan of generic utility files. It tends to grow indefinitely and become hard to maintain.

Comment (trongtt) [inline extension/webapp/src/main/webapp/WEB-INF/gatein-resources.xml]: Ok, I renamed it to `presence`

**Comment by MrHanachoo** [comment]
merged into exodev/develop

**Pull request closed without merging**